src/Microsoft.AspNetCore.Razor.TagHelpers.Testing.Sources/CaseSensitiveTagHelperAttributeComparer.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNetCore.Html;
+
+namespace Microsoft.AspNetCore.Razor.TagHelpers.Testing
+{
+    internal class CaseSensitiveTagHelperAttributeComparer : IEqualityComparer<TagHelperAttribute>
+    {
+        public readonly static CaseSensitiveTagHelperAttributeComparer Default =
+            new CaseSensitiveTagHelperAttributeComparer();
+
+        private CaseSensitiveTagHelperAttributeComparer()
+        {
+        }
+
+        public bool Equals(TagHelperAttribute attributeX, TagHelperAttribute attributeY)
+        {
+            if (attributeX == attributeY)
+            {
+                return true;
+            }
+
+            // Normal comparer (TagHelperAttribute.Equals()) doesn't care about the Name case, in tests we do.
+            return attributeX != null &&
+                string.Equals(attributeX.Name, attributeY.Name, StringComparison.Ordinal) &&
+                attributeX.ValueStyle == attributeY.ValueStyle &&
+                (attributeX.ValueStyle == HtmlAttributeValueStyle.Minimized ||
+                 string.Equals(GetString(attributeX.Value), GetString(attributeY.Value)));
+        }
+
+        public int GetHashCode(TagHelperAttribute attribute)
+        {
+            // Manually combine hash codes here. We can't reference HashCodeCombiner because we have internals visible
+            // from Mvc.Core and Mvc.TagHelpers; both of which reference HashCodeCombiner.
```

Comment:
Is this still true? If yes, clarify who "we" is because Mvc.Core and Mvc.TagHelpers aren't in this repo. If no, use the `HashCodeCombiner`.